apps/server/src/modules/learnroom/controller/api-test/course.api.spec.ts

```diff
@@ -40,39 +40,94 @@ describe('Course Controller (API)', () => {
 		await app.close();
 	});
 
-	const setup = () => {
-		const roles = roleFactory.buildList(1, { permissions: [Permission.COURSE_EDIT] });
-		const user = userFactory.build({ roles });
+	describe('[GET] /courses/', () => {
+		const setup = () => {
+			const roles = roleFactory.buildList(1, { permissions: [Permission.COURSE_EDIT] });
+			const user = userFactory.build({ roles });
 
-		return user;
-	};
+			return { user };
+		};
+		it('should find courses', async () => {
+			const { user: student } = setup();
+			const course = courseFactory.build({ name: 'course #1', students: [student] });
+			await em.persistAndFlush(course);
+			em.clear();
 
-	it('[FIND] courses', async () => {
-		const student = setup();
-		const course = courseFactory.build({ name: 'course #1', students: [student] });
-		await em.persistAndFlush(course);
-		em.clear();
+			currentUser = mapUserToCurrentUser(student);
 
-		currentUser = mapUserToCurrentUser(student);
+			const response = await request(app.getHttpServer()).get('/courses');
 
-		const response = await request(app.getHttpServer()).get('/courses');
+			expect(response.status).toEqual(200);
+			const body = response.body as CourseMetadataListResponse;
+			expect(typeof body.data[0].title).toBe('string');
+		});
+	});
+
+	describe('[GET] /courses/:id', () => {
+		const setup = () => {
+			const roles = roleFactory.buildList(1, { permissions: [Permission.COURSE_EDIT] });
+			const user = userFactory.build({ roles });
+			const students = userFactory.buildList(2);
```

Comment:
is a bit confusing to have a list of roles, but only one role actually
why not call it role and call build instead of build list
also, the user can be called teacher instead. We are using students, why not teacher too?